server/src/main/java/org/opensearch/action/admin/indices/view/CreateViewAction.java

```diff
@@ -0,0 +1,284 @@
+            "create_view_request",
+            args -> new Request((String) args[0], (String) args[1], (List<Target>) args[2])
+        );
+
+        static {
+            PARSER.declareString(ConstructingObjectParser.constructorArg(), View.NAME_FIELD);
+            PARSER.declareString(ConstructingObjectParser.optionalConstructorArg(), View.DESCRIPTION_FIELD);
+            PARSER.declareObjectArray(ConstructingObjectParser.constructorArg(), (p, c) -> Target.fromXContent(p), View.TARGETS_FIELD);
+        }
+
+        public static Request fromXContent(final XContentParser parser) throws IOException {
+            return PARSER.parse(parser, null);
+        }
+    }
+
+    /** Response for view creation */
+    @ExperimentalApi
+    public static class Response extends ActionResponse implements ToXContentObject {
+
+        private final View createdView;
+
+        public Response(final View createdView) {
+            this.createdView = createdView;
+        }
+
+        public Response(final StreamInput in) throws IOException {
+            super(in);
+            this.createdView = new View(in);
+        }
+
+        @Override
+        public void writeTo(final StreamOutput out) throws IOException {
+            this.createdView.writeTo(out);
+        }
+
+        @Override
+        public XContentBuilder toXContent(final XContentBuilder builder, final Params params) throws IOException {
+            builder.startObject();
+            builder.field("view", createdView);
+            builder.endObject();
+            return builder;
+        }
+
+        private static final ConstructingObjectParser<Response, Void> PARSER = new ConstructingObjectParser<>(
+            "create_view_response",
+            args -> new Response((View) args[0])
+        );
+        static {
+            PARSER.declareObject(ConstructingObjectParser.constructorArg(), View.PARSER, new ParseField("view"));
+        }
+
+        public static Response fromXContent(final XContentParser parser) throws IOException {
+            return PARSER.parse(parser, null);
+        }
+    }
+
+    /**
+     * Transport Action for creating a View
+     */
+    public static class TransportAction extends TransportClusterManagerNodeAction<Request, Response> {
+
+        private final ViewService viewService;
+
+        @Inject
+        public TransportAction(
+            final TransportService transportService,
+            final ClusterService clusterService,
+            final ThreadPool threadPool,
+            final ActionFilters actionFilters,
+            final IndexNameExpressionResolver indexNameExpressionResolver,
+            final ViewService viewService
+        ) {
+            super(NAME, transportService, clusterService, threadPool, actionFilters, Request::new, indexNameExpressionResolver);
+            this.viewService = viewService;
+        }
+
+        @Override
+        protected String executor() {
+            return ThreadPool.Names.MANAGEMENT;
```

Comment:
qq, why choose MANAGEMENT threadpool here from all [available threadpools](https://github.com/opensearch-project/OpenSearch/blob/main/server/src/main/java/org/opensearch/threadpool/ThreadPool.java)? Is it because this is cluster manager operation?

Reply:
Yup, since its updating metadata on the cluster I've put it into that threadpool